src/drivers/power_monitor/ina226/ina226_main.cpp

```diff
@@ -0,0 +1,298 @@
+
+#include <px4_platform_common/px4_config.h>
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/module.h>
+#include <drivers/device/i2c.h>
+#include <lib/parameters/param.h>
+#include <lib/perf/perf_counter.h>
+#include <drivers/drv_hrt.h>
+#include <uORB/PublicationMulti.hpp>
+#include <uORB/topics/power_monitor.h>
+#include <px4_platform_common/px4_work_queue/ScheduledWorkItem.hpp>
+
+#include "ina226.h"
+
+#define MAX_I2C_BATTERY_COUNT 2
+
+extern "C" __EXPORT int ina226_main(int argc, char *argv[]);
+
+/**
+ * Local functions in support of the shell command.
+ */
+namespace ina226
+{
+
+INA226	*g_dev[MAX_I2C_BATTERY_COUNT] = {nullptr};
+
+int 	start(int i2c_bus, int address);
+int 	start_bus(int i2c_bus, int address, int battery_index, bool force);
+int 	stop(int i2c_bus, int address);
+int 	info(int i2c_bus, int address);
+
+/**
+ * If a device is already running on the specified bus and address, return the index of that device.
+ * If not, return the index of the first empty slot (nullptr) in the array.
+ * @param i2c_bus The bus, as an index in i2c_bus_options
+ * @param address I2C address of the power monitor
+ * @return If there is already a device running on the given bus with the given address: Return the index of that device
+ *         If there is not already a device running: Return the index of the first nullptr in the array.
+ *         If there are no empty slots in the array, return -1.
+ */
+int get_index(int i2c_bus, int address)
+{
+	int first_nullptr = -1;
+
+	for (size_t i = 0; i < MAX_I2C_BATTERY_COUNT; i++) {
+		//PX4_INFO("Checking number %lu", i);
+		if (g_dev[i] == nullptr) {
+			if (first_nullptr < 0) {
+				first_nullptr = i;
+			}
+
+		} else if (g_dev[i]->get_device_bus() == i2c_bus_options[i2c_bus] && g_dev[i]->get_device_address() == address) {
+			return i;
+		}
+	}
+
+	return first_nullptr;
+}
+
+/**
+ *
+ * Attempt to start driver on all available I2C busses.
+ *
+ * This function will return as soon as the first sensor
+ * is detected on one of the available busses or if no
+ * sensors are detected.
+ *
+ */
+int
+start()
+{
+	for (unsigned i2c_bus = 0; i2c_bus < NUM_I2C_BUS_OPTIONS; i2c_bus++) {
+		int index = get_index(i2c_bus, INA226_BASEADDR);
+
+		if (g_dev[index] == nullptr && start_bus(i2c_bus, INA226_BASEADDR, 1, false) == PX4_OK) {
```

Comment:
`get_index` might return -1. Can you check for that (it's wrong at other places too)?